### PR TITLE
"azurerm_data_factory_pipeline" - supports "concurrency", "elapsed_time_metric_duration" and "run_dimension"

### LINKED ISSUE #11506: Support for concurrency property in azurerm_data_factory_pipeline
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

The azurerm_data_factory_pipeline does not have a "concurrency" property implemented, allowing the maximum concurrency to be set. 

It is available as a property in the data factory UI, and json as below

```json
{
    "name": "pipeline1",
    "properties": {
        "concurrency": 1,
    }
}
```

### New or Affected Resource(s)

* azurerm_data_factory_pipeline

### Potential Terraform Configuration

```hcl
resource "azurerm_data_factory_pipeline" "test" {
  name                = "my-data-factory"
  resource_group_name = azurerm_resource_group.test.name
  data_factory_name   = azurerm_data_factory.test.name
  concurrency = 1
}
```

### References

* .NET API Reference: https://docs.microsoft.com/en-us/dotnet/api/microsoft.azure.management.datafactory.models.pipelineresource.concurrency?view=azure-dotnet
--->

### PR DESCRIPTION
**Pull request by njuCZ**
fix #11506

"run_dimension" is a key value pair, but the value could be a string or an expression type, so made it TypeList. It's used by ChainingTrigger.

ChainingTrigger allows the referenced pipeline to depend on other pipeline runs based on runDimension Name/Value pairs. Upstream pipelines should declare the same runDimension Name and their runs should have the values for those runDimensions. The referenced pipeline run would be triggered if the values for the runDimension match for all upstream pipeline runs.

image

### CODE REVIEW

File: website/docs/r/data_factory_pipeline.html.markdown [hunk 1]
```diff
@@ -72,6 +72,10 @@ The following arguments are supported:
 
 * `annotations` - (Optional) List of tags that can be used for describing the Data Factory Pipeline.
 
+* `concurrency` - (Optional) The max number of concurrent runs for the Data Factory Pipeline. Must be between `1` and `50`.
+
+* `elapsed_time_metric_duration` - (Optional) The TimeSpan value after which an Azure Monitoring Metric is fired.
```
Comment (katbyte) [inline website/docs/r/data_factory_pipeline.html.markdown:77]: this property doesn't match the description
```suggestion
* `moniter_metrics_after_duration` - (Optional) The TimeSpan value after which an Azure Monitoring Metric is fired.
```

File: website/docs/r/data_factory_pipeline.html.markdown [hunk 2]
```diff
@@ -80,6 +84,20 @@ The following arguments are supported:
...
+* `run_dimension` - (Optional) One or more `run_dimension` blocks as defined below.
+
+---
+
+A `run_dimension` block exports the following:
+
+* `name` - The name of run_dimension.
+
+* `value` - The value of run_dimension.
+
```
Comment (katbyte) [inline website/docs/r/data_factory_pipeline.html.markdown:96]: could we not have a _ in the sentance?
```suggestion
* `name` - The name of the run dimension.

* `value` - The value of the run dimension.

```

File: website/docs/r/data_factory_pipeline.html.markdown [hunk 3]
```diff
@@ -80,6 +84,20 @@ The following arguments are supported:
...
+* `dynamic_value_enabled` - is dynamic expression enabled for value?
```
Comment (katbyte) [inline website/docs/r/data_factory_pipeline.html.markdown:97]: what is a dynamic value?

File: website/docs/r/data_factory_pipeline.html.markdown [hunk 4]
```diff
@@ -80,6 +84,20 @@ The following arguments are supported:
 
 * `activities_json` - (Optional) A JSON object that contains the activities that will be associated with the Data Factory Pipeline.
 
+* `run_dimension` - (Optional) One or more `run_dimension` blocks as defined below.
```
Comment (katbyte) [inline website/docs/r/data_factory_pipeline.html.markdown:87]: what is a run dimension?

**Review by katbyte — CHANGES_REQUESTED** [review]

**Comment by ms-henglu** [comment]
Hi @katbyte , I contacted with service team and removed `runDimension` property. Could you please take another look? Thanks!

**Review by katbyte — APPROVED** [review]
Thanks @njuCZ - LGTM 🚀

**Pull request merged**